R/xml_nodes_to_lints.R

```diff
@@ -0,0 +1,86 @@
+#' Convert an XML node or nodeset into a Lint
+#'
+#' Convenience function for converting nodes matched by XPath-based
+#'   linter logic into a [Lint()] object to return.
+#'
+#' @inheritParams lint-s3
+#' @param xml An `xml_node` object (to generate one `Lint`) or an
+#'   `xml_nodeset` object (to generate several `Lint`s), e.g. as returned by
+#'   [xml2::xml_find_all()] or [xml2::xml_find_first()].
+#' @param source_expression A source expression object, e.g. as
+#'   returned typically by [lint()], or more generally
+#'   by [get_source_expressions()].
+#' @param lint_message The message to be included as the `message`
+#'   to the `Lint` object. If `lint_message` is a `function`,
+#'   this function is first applied to `xml` (so it should be a
+#'   function taking an `xml_node` as input and must produce a
+#'   length-1 character as output). If `lint_message` is a character vector the same length as `xml`,
+#'   the `i`-th lint will be given the `i`-th message.
+#' @param column_number_xpath XPath expression to return the column number location of the lint.
+#'   Defaults to the start of the expression matched by `xml`.
+#' @param range_start_xpath XPath expression to return the range start location of the lint.
+#'   Defaults to the start of the expression matched by `xml`.
+#' @param range_end_xpath XPath expression to return the range end location of the lint.
+#'   Defaults to the end of the expression matched by `xml`.
+#' @return For `xml_node`s, a `lint`. For `xml_nodeset`s, `lints` (a list of `lint`s).
+#' @export
+xml_nodes_to_lints <- function(xml, source_expression, lint_message,
+                               type = c("style", "warning", "error"),
+                               column_number_xpath = "number(./@col1)",
+                               range_start_xpath = "number(./@col1)",
```

Comment:
let's set range_start_xpath=column_number_xpath by default, that way they change in concert instead of user needing to write the path twice

Reply:
Sounds good, but I would swap the role (`column_number_xpath = range_start_xpath`)